src/Functions/FunctionsHashing.cpp

```diff
@@ -41,5 +41,22 @@ REGISTER_FUNCTION(Hashing)
     factory.registerFunction<FunctionXxHash64>();
 
     factory.registerFunction<FunctionWyHash64>();
+
+
+#if USE_BLAKE3
```

Comment:
I think it's better to register the function anyway but throw an exception `SUPPORT_IS_DISABLED` with a description why it's disabled without too much details (i.e. what we lack of: rust, blake3 is disabled or maybe it's aarch64). Because if you don't register the function an user will see an exception like "Function is not found" which is not very helpful in this case.

Reply:
Opened #42073 to implement this